### ISSUE TITLE
problem activating plugins with roots.io Bedrock

### ISSUE THREAD

**Issue by slimzc** [gdidentity/on-demand-revalidation#5]
Hi, when I install the plugin using composer with bedrock but when I try to activate, this error is throwed

Fatal error: Uncaught Error: Call to undefined function get_plugin_data() in /var/www/html/web/app/plugins/on-demand-revalidation/on-demand-revalidation.php:118 Stack trace: #0 /var/www/html/web/app/plugins/on-demand-revalidation/on-demand-revalidation.php(50): OnDemandRevalidation->setup_constants() #1 /var/www/html/web/app/plugins/on-demand-revalidation/on-demand-revalidation.php(223): OnDemandRevalidation::instance() #2 /var/www/html/web/wp/wp-settings.php(447): include_once('/var/www/html/w...') #3 /var/www/html/web/wp-config.php(9): require_once('/var/www/html/w...') #4 /var/www/html/web/wp/wp-load.php(55): require_once('/var/www/html/w...') #5 /var/www/html/web/wp/wp-blog-header.php(13): require_once('/var/www/html/w...') #6 /var/www/html/web/wp/index.php(17): require('/var/www/html/w...') #7 {main} thrown in /var/www/html/web/app/plugins/on-demand-revalidation/on-demand-revalidation.php on line 118

obviusly if I add the plugin.php file, everything work normally, but I could not find why the error occur.

Can you help me to resolve it?

**Comment by slimzc** [comment]
Hi, I found a note here: https://developer.wordpress.org/reference/functions/get_plugin_data/#user-contributed-notes

the last contribution say that get_plugin_data(..) is NOT available by default. Maybe this plugin need to add it before use ther plugin.php functions.

using something like : 

```php
if (!function_exists('get_plugin_data')) {
    require_once(ABSPATH . 'wp-admin/includes/plugin.php');
}
```

**Comment by slimzc** [comment]
To fix temporaly, I create a new plugin that only add the inlcudes/plugin.php file. This way, when you plugin is activated, the functions required are load.

**Comment by bebjakub** [comment]
Thank you @slimzc new version with your fix released #6
